config/configtls/configtls.go

```diff
@@ -211,6 +225,24 @@ func (c Config) Validate() error {
 	return nil
 }
 
+func (c ClientConfig) Validate() error {
+	if c.Insecure && !c.hasCA() {
+		return nil
+	}
+	return c.Config.validate()
+}
+
+func (c ServerConfig) Validate() error {
+	if err := c.Config.validate(); err != nil {
+		return err
+	}
```

Comment:
Follow up from the previous comment: we no longer need the recursive call or the `Insecure` check.

```suggestion
func (c ServerConfig) Validate() error {
```